integrated_channels/sap_success_factors/exporters/learner_data.py

```diff
@@ -68,3 +71,40 @@ def get_learner_data_records(self, enterprise_enrollment, completed_date=None, g
                 'No learner data was sent for user [%s] because an SAP SuccessFactors user ID could not be found.',
                 enterprise_enrollment.enterprise_customer_user.username
             )
+
+
+class SapSuccessFactorsLearnerManger(object):
+    """
+    Class to manage SAPSF learners data and their relation with enterprise.
+    """
+
+    def __init__(self, enterprise_configuration, client=SAPSuccessFactorsAPIClient):
+        """
+        Use the ``SAPSuccessFactorsAPIClient`` for content metadata transmission to SAPSF.
+
+        Arguments:
+            enterprise_configuration (required): SAPSF configuration connecting an enterprise to an integrated channel.
+            client: The REST API client that will fetch data from integrated channel.
+        """
+        self.enterprise_configuration = enterprise_configuration
+        self.client = client(enterprise_configuration) if client else None
+
+    def unlink_learners(self):
+        """
+        Iterate over each learner and unlink inactive SAP channel learners.
+
+        This method iterates over each enterprise learner and unlink learner
+        from the enterprise if the learner is marked inactive in the related
+        integrated channel.
+        """
+        sap_inactive_learners = self.client.get_inactive_sap_learners()
+        enterprise_customer = self.enterprise_configuration.enterprise_customer
+        tpa_provider = enterprise_customer.enterprise_customer_identity_provider.provider_id
+        for sap_inactive_learner in sap_inactive_learners:
+            social_auth_user = get_user_from_social_auth(tpa_provider, sap_inactive_learner['studentID'])
+            if social_auth_user:
+                # Unlink user email from related Enterprise Customer
+                EnterpriseCustomerUser.objects.unlink_user(
```

Comment:
I'm assuming that we already write an informative log entry if the `unlink_user` operation fails for some reason.

Reply:
I have added the exception handling for the possible exceptions `(EnterpriseCustomerUser.DoesNotExist, PendingEnterpriseCustomerUser.DoesNotExist)`.